### PR TITLE
Qwatch integration tests with where condition and benchmarking

### LINKED ISSUE #188: Add integration test for testing WHERE clause in QWATCH.
WHERE clause support was added to DSQL in #187. However, we don't have any integration test cases validating its behavior.

## Requirements
- Add test cases that validate E2E flow for QWATCH queries using the WHERE clause.
- See [tests/qwatch_test.go](https://github.com/DiceDB/dice/blob/master/tests/qwatch_test.go) for reference.
- benchmark the code and measure the time taken and memory allocs using [benchmem](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.
- Add integration test results as a screenshot in the PR description (since we don't run integration tests on the CI for now)

### PR DESCRIPTION
**Pull request by deep-adeshraa**
fixes #188 
Integration test case for Qwatch with where condition and benchmarking for Qwatch with SET commands.

### CODE REVIEW

**Comment by deep-adeshraa** [comment]
@JyotinderSingh Benchmarking QWATCH with large number of requests fails the I/O test cases. I have commented out benchmarking code. If you could guide me to navigate the bug, I will be happy to solve it.

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh Benchmarking QWATCH with large number of requests fails the I/O test cases. I have commented out benchmarking code. If you could guide me to navigate the bug, I will be happy to solve it.

We are making some changes to the QWATCH feature. It might help with some of these issues.
We can re-explore this over the weekend (since the qwatch changes would be merged by then)

**Comment by JyotinderSingh** [comment]
Please rebase the PR, we just had a large change go in which might've introduced conflicts.

**Comment by JyotinderSingh** [comment]
@deep-adeshraa are you still working on this?

**Comment by deep-adeshraa** [comment]
@JyotinderSingh give me this weekend last - I will finish it up with new code you guys added

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh give me this weekend last - I will finish it up with new code you guys added

Any updates?

**Comment by psrvere** [comment]
@JyotinderSingh - if there is inactivity on this issue, I can pick this up.

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh - if there is inactivity on this issue, I can pick this up.

Feel free to take this up.

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh - if there is inactivity on this issue, I can pick this up.

Please comment on the associated issue so it can be assigned to you.

**Comment by lucifercr07** [comment]
@deep-adeshraa can you please resolve the conflicts.

File: tests/qwatch_test.go [hunk 1]
```diff
@@ -491,3 +516,66 @@ func cleanupJSONOrderByKeys(publisher net.Conn) {
...
+				v, err := rp.DecodeOne()
+				assert.NilError(t, err)
+				update := v.([]interface{})
+				assert.DeepEqual(t, expectedUpdate, update[len(update)-1])
+			}
+		}
+	}
+}
+
+// func BenchmarkQwatchCommandSingleQuery(b *testing.B) {
```
Comment (lucifercr07) [inline tests/qwatch_test.go:566]: Please remove the code block if not in use.

**Pull request closed without merging**